.github/actions/test/action.yml

```diff
@@ -16,9 +16,18 @@ runs:
     - run: npm run test:unit
       name: Unit Tests
       shell: bash
-    - run: npm run test:miniflare
-      name: Miniflare Tests
-      shell: bash
     - run: npm run test:integration
       name: Integration Tests
       shell: bash
+    - run: |
+        # If the Miniflare tests fail we need to capture the output and check for the error,
+        # because the error is not bubbled up from miniflare and the tests will pass.
+        npm run test:miniflare | tee test-output.log
```

Comment:
**suggestion (non-blocking):** Can we get `npm run test:miniflare` itself to `exit 1`? Ideally it shouldn't cover up failures at the command line either. If that's too complex for right now, let's add a card for it in the backlog.